### PR TITLE
added ref_fasta, prefix, exclude_pg, template_coordinate, and verbosi…

### PR DESCRIPTION
**Pull request by mchenaux**
…ty flags

### CODE REVIEW

File: q2_samtools/plugin_setup.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 """QIIME 2 plugin for samtools."""
 
 import qiime2.plugin
+from q2_types.feature_data import FeatureData, RNASequence, Sequence
```
Comment (HenziKou) [inline q2_samtools/plugin_setup.py:4]: Remove unused import `RNASequence`

File: q2_samtools/_samtools.py [hunk 2]
```diff
@@ -1,6 +1,13 @@
 import os
 import subprocess
+from typing import Union
 
+from q2_types.feature_data._format import (
+    DNAFASTAFormat,
+    DNASequencesDirectoryFormat,
+    RNAFASTAFormat,
+    RNASequencesDirectoryFormat,
```
Comment (HenziKou) [inline q2_samtools/_samtools.py:9]: Remove unused imports `DNASequencesDirectoryFormat` and `RNASequencesDirectoryFormat`

File: q2_samtools/_samtools.py [hunk 3]
```diff
@@ -31,15 +43,25 @@ def sort(
             str(compression_level),
             "-m",
             str(memory_per_thread),
+            "-T",
+            str(prefix),
```
Comment (HenziKou) [inline q2_samtools/_samtools.py:47]: What will happen if we pass the `-T` flag with the default value set as an empty string? Will samtools throw an error? If an error is thrown then this needs to be moved into its own conditional block.

Comment (mchenaux) [reply q2_samtools/_samtools.py:47]: Ok, I think I need some help figuring out the desired behavior here.

File: q2_samtools/_samtools.py [hunk 4]
```diff
@@ -33,13 +40,23 @@ def sort(
...
         if name_sort:
             cmd.append("-n")
+        if exclude_pg:
+            cmd.append("--no-PG")
+        if template_coordinate:
+            cmd.append("--template-coordinate")
+        if prefix:
+            cmd.extend(["-T", prefix])
         if tag_sort:
             cmd.extend(["-t", tag_sort])
```
Comment (HenziKou) [inline q2_samtools/_samtools.py:55]: We should probably wrap the arguments for prefix and tag_sort in a string class, e.g. `str(prefix)`

**Review by HenziKou — CHANGES_REQUESTED** [review]

**Review by HenziKou — APPROVED** [review]

**Pull request merged**